Examples/Algorithms/TrackFinding/src/AmbiguityResolutionAlgorithm.cpp

```diff
@@ -38,148 +38,145 @@ ActsExamples::AmbiguityResolutionAlgorithm::AmbiguityResolutionAlgorithm(
-              .template get<ActsExamples::IndexSourceLink>()
-              .index();
+      std::vector<std::size_t> measurements;
+      traj.multiTrajectory().visitBackwards(tip, [&](const auto& hit) {
+        if (hit.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
+          std::size_t iMeasurement =
+              hit.getUncalibratedSourceLink()
+                  .template get<ActsExamples::IndexSourceLink>()
+                  .index();
+          measurements.push_back(iMeasurement);
+        }
+        return true;
+      });
 
-      ++hitCountPerMeasurement[indexHit];
+      ++state.numberOfTracks;
 
-      return true;
-    });
-  }
+      state.trackTips.emplace_back(iTraj, tip);
+      state.trackChi2.push_back(trajState.chi2Sum / trajState.NDF);
+      state.trackParameters.push_back(traj.trackParameters(tip));
+      state.measurementsPerTrack.push_back(std::move(measurements));
 
-  std::vector<std::size_t> sharedHitCountPerTrack(trackIndices.size(), 0);
+      state.selectedTracks.insert(iTrack);
 
-  for (std::size_t i = 0; i < trackIndices.size(); ++i) {
-    const auto indexTrack = trackIndices[i];
-    const auto [indexTraj, tip] = trackTips[indexTrack];
-    const auto& traj = trajectories[indexTraj];
+      ++iTrack;
+    }
+  }
 
-    traj.multiTrajectory().visitBackwards(tip, [&](const auto& state) {
-      if (!state.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
-        return true;
-      }
+  for (std::size_t iTrack = 0; iTrack < state.numberOfTracks; ++iTrack) {
+    for (auto iMeasurement : state.measurementsPerTrack[iTrack]) {
+      state.tracksPerMeasurement[iMeasurement].insert(iTrack);
+    }
+  }
 
-      const std::size_t indexHit =
-          state.getUncalibratedSourceLink()
-              .template get<ActsExamples::IndexSourceLink>()
-              .index();
+  state.sharedMeasurementsPerTrack =
+      std::vector<std::size_t>(state.trackTips.size(), 0);
 
-      if (hitCountPerMeasurement[indexHit] > 1) {
-        ++sharedHitCountPerTrack[i];
+  for (std::size_t iTrack = 0; iTrack < state.numberOfTracks; ++iTrack) {
+    for (auto iMeasurement : state.measurementsPerTrack[iTrack]) {
+      if (state.tracksPerMeasurement[iMeasurement].size() > 1) {
+        ++state.sharedMeasurementsPerTrack[iTrack];
       }
-
-      return true;
-    });
+    }
   }
 
-  return sharedHitCountPerTrack;
+  return state;
 }
 
-std::size_t computeTrackHits(
-    const Acts::ConstVectorMultiTrajectory& multiTrajectory,
-    const std::size_t tip) {
-  std::size_t result = 0;
+void removeTrack(State& state, std::size_t iTrack) {
+  for (auto iMeasurement : state.measurementsPerTrack[iTrack]) {
+    state.tracksPerMeasurement[iMeasurement].erase(iTrack);
 
-  multiTrajectory.visitBackwards(tip, [&](const auto&) { ++result; });
+    if (state.tracksPerMeasurement[iMeasurement].size() == 1) {
+      auto jTrack = *std::begin(state.tracksPerMeasurement[iMeasurement]);
```

Comment:
```suggestion
      auto jTrack = state.tracksPerMeasurement[iMeasurement].front();
```
?

Reply:
this does not work on the set unfortunately

Reply:
Ok then nevermind.